docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -8,48 +8,141 @@ title: Authorization
+the following format:
+
+```http
+WWW-Authenticate: Bearer resource_metadata="https://resource.example.com/.well-known/oauth-protected-resource",
+     scope="mcp:read mcp:write"
+```
+
+A MCP server **MUST** host its Protected Resource Metadata (PRM) document, as outlined in
+[Section 3 of the RFC draft](https://www.ietf.org/archive/id/draft-ietf-oauth-resource-metadata-13.html#section-3).
+
+A MCP server **MUST** return the minimum required set of scopes in the `scope` parameter within the
+`WWW-Authenticate` header in the initial handshake as well as any additional scopes in subsequent responses
+that require incremental authorization.
+
+A MCP server **MUST** include its AS configuration in the PRM document as outlined in
+[Section 2 of the RFC draft](https://www.ietf.org/archive/id/draft-ietf-oauth-resource-metadata-13.html#section-2).
+
+A MCP client **MUST** issue a `GET` request to the URL in `resource_metadata` to get information about
+the AS.
 
-2. MCP auth implementations **SHOULD** support the OAuth 2.0 Dynamic Client Registration
-   Protocol ([RFC7591](https://datatracker.ietf.org/doc/html/rfc7591)).
+A MCP client **MUST** verify that the `resource` value in the PRM document matches the MCP server
+protocol and fully-qualified domain name (FQDN).
 
-3. MCP servers **SHOULD** and MCP clients **MUST** implement OAuth 2.0 Authorization
-   Server Metadata ([RFC8414](https://datatracker.ietf.org/doc/html/rfc8414)). Servers
-   that do not support Authorization Server Metadata **MUST** follow the default URI
-   schema.
+A MCP client **SHOULD** evaluate or permit the user to evaluate the trustworthiness of AS
+discovered through the PRM document to prevent phishing. For example, an MCP client could prompt 
+the user "Do you want to login with `https://authserver.example`?"
 
-### 2.1.1 OAuth Grant Types
+A MCP client **MUST** use `authorization_servers` values from the PRM document to construct
+the OAuth 2.0 metadata discovery document path, following
+[RFC 8414](https://datatracker.ietf.org/doc/html/rfc8414#section-3).
+
+A MCP client **MAY** use the AS-provided registration endpoint to register itself with the AS.
+
+A MCP client **MAY** be pre-registered with an AS and not require dynamic registration.
+
+A MCP client **MUST** send the `resource` identifier to the AS, following
+[RFC 8707](https://www.rfc-editor.org/rfc/rfc8707.html).
+
+A MCP client **MAY** implement Demonstrating Proof of Possession (DPoP) and use it to the extent
+supported by the AS. See
+[RFC 9449](https://datatracker.ietf.org/doc/html/rfc9449).
+
+A MCP clients **MUST** initiate an
+[OAuth 2.1 (draft)](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12) authorization
+flow after receiving a `HTTP 401 Unauthorized` response.
+
+### 2.2.1 OAuth Grant Types
 
 OAuth specifies different flows or grant types, which are different ways of obtaining an
 access token. Each of these targets different use cases and scenarios.
```

Comment:
This is intended for the section below, which I can't tie my comment to.
> MCP servers **SHOULD** support the OAuth grant types that best align with the intended
audience. For instance:

Is this still valid given that MCP servers are now resource servers, which are generally not concerned with the authorization grant type?